aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -852,11 +852,17 @@ private boolean modelExists(Model model, QueryPredicate predicate) throws DataSt
      * @return the Model instance from SQLite, if it exists, otherwise null.
      */
     private Model query(Model model) {
+        final String modelName = getModelName(model);
+        final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+        final SQLiteTable table = SQLiteTable.fromSchema(schema);
+        final String primaryKeyName = table.getPrimaryKeyColumnName();
+        final QueryPredicate matchId = QueryField.field(primaryKeyName).eq(model.getId());
+
         Iterator<? extends Model> result = Single.<Iterator<? extends Model>>create(emitter -> {
             if (model instanceof SerializedModel) {
-                query(getModelName(model), Where.id(model.getId()), emitter::onSuccess, emitter::onError);
+                query(modelName, Where.matches(matchId), emitter::onSuccess, emitter::onError);
             } else {
-                query(model.getClass(), Where.id(model.getId()), emitter::onSuccess, emitter::onError);
+                query(model.getClass(), Where.matches(matchId), emitter::onSuccess, emitter::onError);
```

Comment:
How are they different, internally?

Reply:
`Where.id` uses hardcoded `"id"` as column name when building query statement, which causes ambiguous column name SQLiteException. `SQLiteColumn#getPrimaryKeyColumnName()` appends column name to the table name like following: `<model>.<column>`.